whisper.cpp

```diff
@@ -7196,14 +7204,19 @@ void whisper_log_set(ggml_log_callback log_callback, void * user_data) {
     g_state.log_callback_user_data = user_data;
 }
 
-GGML_ATTRIBUTE_FORMAT(2, 3)
-static void whisper_log_internal(ggml_log_level level, const char * format, ...) {
+GGML_ATTRIBUTE_FORMAT(5, 6)
+static void whisper_log_internal(ggml_log_level level, const char * file, const char * func, int line, const char * format, ...) {
     va_list args;
     va_start(args, format);
     char buffer[1024];
-    int len = vsnprintf(buffer, 1024, format, args);
-    if (len < 1024) {
+    int len_prefix = snprintf(buffer, 1024, "[%s, %d]: ", func, line); // param file not used in this file
+    int len = vsnprintf(buffer + len_prefix, 1024 - len_prefix, format, args);
```

Comment:
if `snprintf` fails due to an output error then this will write to an address before the buffer's start.